### PR TITLE
PXP-635: [CLI] Terminate the TUI when there is a panic on TUI

### PR DESCRIPTION
**Pull request by mfauzaan**
## Summary
Currently, when a panic occurs in the Wukong CLI, the panic message is displayed on the TUI screen and breaks the TUI rendering and makes it TUI unusable. 

To address this issue, a panic hook has been implemented to capture TUI errors and ensure a safe exit. Additionally, network panics needs to be handled on a case-by-case basis, providing relevant error messages to users in the corresponding panel.

Ticket: https://mindvalley.atlassian.net/browse/PXP-635

## What's Changed

- [x] Add panic hook

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/174#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cli/src/commands/tui/mod.rs
Comment (jk-gan) [inline cli/src/commands/tui/mod.rs:200]: what does this mean?

Comment (mfauzaan) [reply cli/src/commands/tui/mod.rs:200]: This was originally taken from [here](https://github.com/ClementTsang/bottom/blob/9d0bfbbf93f85907893fd879710eaba44160b236/src/lib.rs#L298)

As my understanding was that its a placeholder to match the string return types

Comment (jk-gan) [reply cli/src/commands/tui/mod.rs:200]: ok got it

**Review by jk-gan — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/174#pullrequestreview-1743772767" title="Approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/174#" title="Author self-approved">mfauzaan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
